discovery-infra/test_infra/helper_classes/entity.py

```diff
@@ -0,0 +1,67 @@
+from abc import ABC, abstractmethod
+from typing import Optional, Union
+
+from assisted_service_client import models
+from logger import log
+from test_infra.assisted_service_api import InventoryClient
+from test_infra.helper_classes.config import BaseEntityConfig
+from test_infra.helper_classes.nodes import Nodes
+
+
+class Entity(ABC):
+    def __init__(self, api_client: InventoryClient, config: BaseEntityConfig, nodes: Optional[Nodes] = None):
+        self._config = config
+        self.api_client = api_client
+        self.nodes: Nodes = nodes
+        self._infra_env = None
+        self.id = self._create()
+
+    @property
+    def entity_name(self):
+        return self.__class__.__name__.lower()
```

Comment:
1. `TerraformController` doesn't have the entity object (but we might want to update the configurations)
2. The entity name I added is different then the value returned by `TerraformController._get_entity_name`

Reply:
But regardless this PR it can be changed to:
```python
    def _get_entity_name(self) -> BaseName:
          ...
   
    # Can be replaced with
    self._entity_config.entity_name

```

Reply:
yeah that's exactly what I'm suggesting :)